packages/documentation-v7/src/stories/components/select/select.stories.ts

```diff
@@ -0,0 +1,328 @@
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    validation: {
+      name: 'Validation',
+      description: 'Controls the validation state appearance of the component.',
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid',
+        },
+      },
+      options: ['null', 'is-valid', 'is-invalid'],
+      table: {
+        category: 'States',
+      },
+    },
+  },
+};
+
+export default meta;
+
+type Story = StoryObj;
+
+const Template: Story = {
+  render: (args: Args, context: StoryContext) => {
+    const [_, updateArgs] = useArgs();
+    const id = `${context.viewMode}_${context.story.replace(/\s/g, '-')}_ExampleSelect`;
+    const classes = ['form-select', args.size, args.validation]
+      .filter(c => c && c !== 'null')
+      .join(' ');
+    const useAriaLabel = !args.floatingLabel && args.hiddenLabel;
+    const label = !useAriaLabel
+      ? html`
+          <label for="${id}" class="form-label">${args.label}</label>
+        `
+      : null;
+    const optionElements = Array.from({ length: args.options - 1 }, (_, i) => i + 2).map(
+      (key: number) => html`
+        <option value="valoro_${key}">Opcion ${key}</option>
+      `,
+    );
+    const options = [
+      html`
+        <option>Elektu opcion...</option>
+      `,
+      ...optionElements,
+    ];
+    const contextuals = [
+      args.validation === 'is-valid'
+        ? html`
+            <p class="valid-feedback">Ggranda sukceso!</p>
+          `
+        : null,
+      args.validation === 'is-invalid'
+        ? html`
+            <p class="invalid-feedback">Eraro okazis!</p>
+          `
+        : null,
+      args.hint !== ''
+        ? html`
+            <div class="form-text">${args.hint}</div>
+          `
+        : null,
+    ];
+    const control = html`
+      <select
+        id=${id}
+        class=${classes}
+        defaultValue=${args.value}
+        ?multiple=${args.multiple}
+        size=${args.multipleSize}
+        ?disabled=${args.disabled}
+        aria-label=${useAriaLabel ? args.label : undefined}
+        aria-invalid=${VALIDATION_STATE_MAP[args.validation]}
+        @change=${(e: Event) => {
+          updateArgs({ value: (e.target as HTMLSelectElement).value });
+
+          if (document.activeElement === e.target) {
+            setTimeout(() => {
+              const element: HTMLSelectElement | null = document.querySelector(`#${id}`);
+              if (element) element.focus();
+            }, 25);
+          }
+        }}
```

Comment:
To reduce complexity, you can extract this function:
- Define the function somewhere beside the `control` definition, like so: `function onChange(e: Event) { ... }`
- Bind the function like so `@change="${onChange}"`.